generators/server/templates/src/main/java/package/config/_WebConfigurer.java

```diff
@@ -53,13 +54,13 @@ public void onStartup(ServletContext servletContext) throws ServletException {
         EnumSet<DispatcherType> disps = EnumSet.of(DispatcherType.REQUEST, DispatcherType.FORWARD, DispatcherType.ASYNC);<% if (clusteredHttpSession == 'hazelcast') { %>
         initClusteredHttpSessionFilter(servletContext, disps);<% } %>
         initMetrics(servletContext, disps);
-        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {
-            initCachingHttpHeadersFilter(servletContext, disps);
-            initStaticResourcesProductionFilter(servletContext, disps);
-        }<% if (devDatabaseType == 'h2Disk' || devDatabaseType == 'h2Memory') { %>
-        if (env.acceptsProfiles(Constants.SPRING_PROFILE_DEVELOPMENT)) {
-            initH2Console(servletContext);
-        }<% } %>
+        if (env.acceptsProfiles(Constants.SPRING_PROFILE_PRODUCTION)) {<% if (!skipClient) { %>
```

Comment:
yes a few lines more in generator is better than generated ugly code. That's why those were put in separate if blocks, you could also do that to avoid lot of if else in generator